src/java.base/share/classes/java/util/jar/JarOutputStream.java

```diff
@@ -81,6 +81,20 @@ public JarOutputStream(OutputStream out) throws IOException {
      * The current time will be used if the entry has no set modification
      * time.
      *
+     * The zip file format does not record the compression level used for the
+     * deflation of its entries. It doesn't even mandate a predefined compression
+     * ratio per compression level.
+     *
+     * The compressed size attribute of a {@code ZipEntry} reflects the compression
+     * level and {@code Deflater} which were active at the time when the
+     * {@code ZipEntry} was created. They might differ from the compression level
+     * and {@code Deflater} of this {@code JarOutputStream}.
+     *
+     * Unless explictely set by calling {@link ZipEntry#setCompressedSize(long)}
+     * this output stream will ignore the compressed size of a {@code ZipEntry}
+     * and re-compute its value automatically after the associted data has been
+     * completely deflated.
+     *
```

Comment:
We probably need to do a bit wordsmithing here.

I think I'd drop the first two sentences and instead start a new paragraph here (<p> tag) with "Unless explicitly set, the output stream will ignore ..." and see how that looks.

Probably should be "the ZipEntry" rather than "a ZipEntry" to be consistent with the parameter description.

Reply:
Alan makes a good point.  Perhaps we keep things simple and focus solely on tweaking the description of the change in behavior which is described in your last paragraph

Reply:
I totally agree. What about using just the last sentence (as you've proposed) in the spec section and add the other to as @implNote? O you think the last sentence will be enough?

Reply:
I think we can just go with the last sentence/paragraph.  Perhaps  we can further simplify the  paragraph/sentence with something like:

The compressed entry size will be recalculated  for compressed (DEFLATED) entries when  ZipEntry::setCompressedSize has not been explicitly called on the ZipEntry.

or 

The compressed (DEFLATED) entry size will be recalculated when  ZipEntry::setCompressedSize has not been explicitly called on the ZipEntry.

Reply:
I think the wording looks much better now